### ISSUE TITLE
Crash on iOS 10.2.1 - [MTLRenderPipelineDescriptorInternal setRasterSampleCount:]: unrecognized selector sent to instance 

### ISSUE THREAD

**Issue by ducle01490** [MetalPetal/MetalPetal#177]
Hello,
Thank you for your useful library. 
I used the latest version of MetalPedal (v 1.11.1) in my app and i got crash on iOS 10.2.1 (iPhone 6s) as below:

```
Fatal Exception: NSInvalidArgumentException
-[MTLRenderPipelineDescriptorInternal setRasterSampleCount:]: unrecognized selector sent to instance 0x1701b2600
--------------------------------------------------
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x1827851b8 __exceptionPreprocess
1  libobjc.A.dylib                0x1811bc55c objc_exception_throw
2  CoreFoundation                 0x18278c268 __methodDescriptionForSelector
3  CoreFoundation                 0x182789270 ___forwarding___
4  CoreFoundation                 0x18268280c _CF_forwarding_prep_0
5  MetalPetal                     0x100b4ccd0 -[MTIRenderPipelineKernel newKernelStateWithContext:configuration:error:] + 178 (MTIRenderPipelineKernel.m:178)
6  MetalPetal                     0x100b30794 -[MTIContext(Internal) kernelStateForKernel:configuration:error:] + 551 (MTIContext.m:551)
7  MetalPetal                     0x100b4d94c -[MTIImageRenderingRecipe resolveWithContext:resolver:error:] + 299 (MTIRenderPipelineKernel.m:299)
8  MetalPetal                     0x100b4ed4c -[MTIImageRenderingPromise resolveWithContext:error:] + 467 (MTIRenderPipelineKernel.m:467)
9  MetalPetal                     0x100b3eafc -[MTIImageRenderingContext resolutionForImage:error:]
10 MetalPetal                     0x100b2cd48 -[MTIContext(Rendering) startTaskToRenderImage:toDrawableWithRequest:error:completion:] + 357 (MTIContext+Rendering.m:357)
11 MetalPetal                     0x100b2b8fc -[MTIContext(Rendering) startTaskToRenderImage:toDrawableWithRequest:error:] + 128 (MTIContext+Rendering.m:128)
12 MetalPetal                     0x100b2b264 -[MTIContext(Rendering) renderImage:toDrawableWithRequest:error:] + 46 (MTIContext+Rendering.m:46)
13 MetalPetal                     0x100b41cfc -[MTIImageView drawInMTKView:] + 207 (MTIImageView.m:207)
14 MetalKit                       0x19402bec0 -[MTKView draw]
15 QuartzCore                     0x185ae47fc CA::Layer::display()
16 QuartzCore                     0x185ada314 CA::Layer::display_if_needed(CA::Transaction*)
17 QuartzCore                     0x185ada398 CA::Layer::layout_and_display_if_needed(CA::Transaction*)
18 QuartzCore                     0x185a573e0 CA::Context::commit_transaction(CA::Transaction*)
19 QuartzCore                     0x185a7ea68 CA::Transaction::commit()
20 QuartzCore                     0x185a7f488 CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*)
21 CoreFoundation                 0x1827320c0 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
22 CoreFoundation                 0x18272fcf0 __CFRunLoopDoObservers
23 CoreFoundation                 0x182730180 __CFRunLoopRun
24 CoreFoundation                 0x18265e2b8 CFRunLoopRunSpecific
25 GraphicsServices               0x184112198 GSEventRunModal
26 UIKit                          0x1886a57fc -[UIApplication _run]
27 UIKit                          0x1886a0534 UIApplicationMain
28 AppTest                        0x10014e778 main + 16 (main.m:16)
29 ???                            0x1816415b8 (Missing)
```

### Checklist
- [x ] I've read the [README](https://github.com/MetalPetal/MetalPetal/blob/master/README.md)
- [x ] I've searched for [existing GitHub issues](https://github.com/MetalPetal/MetalPetal/issues)

**Comment by YuAo** [comment]
Could you try again with the 1.11.2_patch branch, see if the issue is fixed.

**Comment by ducle01490** [comment]
Thank you for quick replying. After updated code from 1.11.2_patch branch, app did not crash.
However i founded new issue when try to run the MetalPetalDemo project then choose Camera on iPad mini 2 (iOS 10.3).
This issue happened when i run code release version (1.11.0 and 1.11.1). I checked code version 1.10.6, it worked without same issue.
Please see the detail:
```
2020-07-16 13:24:50.390995 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:24:50.391171 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:24:53.669470 MetalPetalDemo[2522:421370] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:24:53.669848 MetalPetalDemo[2522:421370] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:24:53.670141 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Discarded (victim of GPU error/recovery) (IOAF code 5)
2020-07-16 13:24:53.670221 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Discarded (victim of GPU error/recovery) (IOAF code 5)
2020-07-16 13:24:56.946551 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:24:56.946620 MetalPetalDemo[2522:421300] Execution of the command buffer was aborted due to an error during execution. Discarded (victim of GPU error/recovery) (IOAF code 5)
2020-07-16 13:24:56.946712 MetalPetalDemo[2522:421300] Execution of the command buffer was aborted due to an error during execution. Discarded (victim of GPU error/recovery) (IOAF code 5)
2020-07-16 13:24:56.946718 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:25:00.204701 MetalPetalDemo[2522:421300] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:25:00.204818 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Discarded (victim of GPU error/recovery) (IOAF code 5)
2020-07-16 13:25:00.204871 MetalPetalDemo[2522:421300] Execution of the command buffer was aborted due to an error during execution. Caused GPU Hang Error (IOAF code 3)
2020-07-16 13:25:00.204897 MetalPetalDemo[2522:421635] Execution of the command buffer was aborted due to an error during execution. Discarded (victim of GPU error/recovery) (IOAF code 5)
```

**Comment by YuAo** [comment]
@ducle01490  I guess it has something to do with the pixel buffer texture bridge issue happens on older devices. https://github.com/MetalPetal/MetalPetal/issues/64

Let me see if I can apply a quick fix.

**Comment by YuAo** [comment]
@ducle01490 Done. See #178

**Comment by ducle01490** [comment]
Thank you @YuAo 
The issue is fixed! it works now!

**Comment by YuAo** [comment]
@ducle01490 New release published: 1.11.2
